### DISCUSSION TITLE
Add syntax to force hide URL preview for a link you're posting

### DISCUSSION THREAD

**Discussion by n-o-o-b** [element-hq/element-meta#877] [Ideas]
Discord has a feature where you can put < > around a link you're posting and it won't show the preview. It would be cool to have a similar feature in Riot.

**Comment by JaniM** [comment]
Force is a strong word. I think this'd be a useful feature, if we changed it to _hint_. Let the sender hint that a link might not need a preview (useful if you're sending a bunch of links, for example), but also add a button for expanding the previews if the recipient wants to do so.

**Comment by t3chguy** [comment]
Echoing Lampholder:

> I found myself wishing for this recently - the GDPR message text from the server notice room is particularly noisy with the URL previews.

> Talking it through with @lukebarnard1 we concluded that perhaps this could be implemented (on the rendering side) by adding a custom attribute to the HTML of the link in the `formattedBody`, something like `data-mx-preview=false`.

> This would allow any clients who submit a formatted body to supress the preview - the question then is how to provide the UX for this. Something like discord's <url> could work, but it doesn't play very nicely with established markdown conventions.

**Comment by t3chguy** [comment]
(bumping prio as per other issue)

**Comment by t3chguy** [comment]
One question remains, is it acceptable that there'll be no way with plaintext (just `body`) to mark a URL as unwanting of a preview?

**Comment by ara4n** [comment]
mixins would be a nice way to solve this :(

**Comment by t3chguy** [comment]
Would they though? You'd need a weird map[url]bool mixin to be able to specify per url in the event

**Comment by ara4n** [comment]
yeah, it wouldn't help per-url, but would help per event.

**Comment by t3chguy** [comment]
this issue seems to be targeted per-url :(

**Comment by Wendell250** [comment]
Was this implemented at some point?
If a URL is encased in backticks it will hide the preview, e.g.: \`https://matrix.org`

![hidden_preview](https://user-images.githubusercontent.com/90041836/133695013-a7b9d87b-0b33-4165-a0d3-853c0c547cb8.png)

Or is this purely client-sided while still making a query on backend?

**Comment by MadLittleMods** [comment]
@Wendell250's solution seems pretty viable. I'm tempted to close this until someone really wants normal styled links without a preview.

The other unsolved case is when you use inline link syntax which also triggers a preview `[do some search](https://www.bing.com/)`

![](https://user-images.githubusercontent.com/558581/134750393-900cfe42-75a0-4f42-812c-669756be2381.png)

**Comment by urmaul** [comment]
Building a matrix bot that sends HTML messages. This feature would let me put many not-important-but-still-useful links there.
